webrender_api/src/display_list.rs

```diff
@@ -1112,21 +1128,21 @@ impl DisplayListBuilder {
         I: IntoIterator<Item = ComplexClipRegion>,
         I::IntoIter: ExactSizeIterator,
     {
-        let parent_id = self.clip_stack.last().unwrap().scroll_node_id;
-        self.define_scroll_frame_with_parent(
+        let scrollinfo = *self.clip_stack.last().unwrap();
+        self.define_scroll_frame_with_scrollinfo(
             id,
-            parent_id,
+            scrollinfo,
             content_rect,
             clip_rect,
             complex_clips,
             image_mask,
             scroll_sensitivity)
     }
 
-    pub fn define_scroll_frame_with_parent<I>(
+    pub fn define_scroll_frame_with_scrollinfo<I>(
```

Comment:
A ClipScrollNode can only have one parent, so I'm not sure I understand the benefit of passing in a ClipScrollInfo here. I think this should probably just accept a ClipId to avoid ambiguity. This would mean these API points would maintain with "with_parent" naming.

Reply:
Ah that would explain why my previous attempt to define the clip chains and scroll chains more independently didn't work. This seems like an inconsistent API, because if you push a rect inside a ClipAndScrollInfo it means one thing (scroll by this clip id, clip by this other clip id) but if you define a clip inside a ClipAndScrollInfo it means something else (scroll by this clip, ignore this other clip id).